src/daft-core/src/array/ops/date.rs

```diff
@@ -107,4 +107,40 @@ impl TimestampArray {
             Int32Array::from((self.name(), Box::new(date_arrow))),
         ))
     }
+
+    pub fn hour(&self) -> DaftResult<UInt32Array> {
+        let physical = self.physical.as_arrow();
+        let DataType::Timestamp(timeunit, tz) = self.data_type() else {
+            unreachable!("Timestamp array must have Timestamp datatype")
+        };
+        let tu = timeunit.to_arrow();
+        let date_arrow = match tz {
+            Some(tz) => match arrow2::temporal_conversions::parse_offset(tz) {
+                Ok(tz) => Ok(arrow2::array::UInt32Array::from_iter(physical.iter().map(
+                    |ts| {
+                        ts.map(|ts| {
+                            arrow2::temporal_conversions::timestamp_to_datetime(*ts, tu, &tz).hour()
+                        })
+                    },
+                ))),
+                Err(e) => Err(DaftError::TypeError(format!(
+                    "Cannot parse timezone in Timestamp datatype: {}, error: {}",
+                    tz, e
+                ))),
+            },
+            None => Ok(arrow2::array::UInt32Array::from_iter(physical.iter().map(
+                |ts| {
+                    ts.map(|ts| {
+                        arrow2::temporal_conversions::timestamp_to_naive_datetime(*ts, tu).hour()
+                    })
+                },
+            ))),
+        }?;
+
+        Ok(UInt32Array::new(
+            std::sync::Arc::new(Field::new(self.name(), DataType::UInt32)),
+            Box::new(date_arrow),
+        )
+        .unwrap_or(UInt32Array::from(("hour", vec![0]))))
```

Comment:
You shouldn't have to unwrap the `UInt32Array::new` and then wrap it in `Ok`. You should be able to return the result from UInt32Array::new directly! 
```
UInt32Array::new(
            std::sync::Arc::new(Field::new(self.name(), DataType::UInt32)),
            Box::new(date_arrow),
        )
```

Reply:
Yea, not sure what I was thinking. Fixing it now.